test/e2e/common/pods.go

```diff
@@ -192,8 +195,30 @@ var _ = framework.KubeDescribe("Pods", func() {
 			LabelSelector:   selector.String(),
 			ResourceVersion: pods.ListMeta.ResourceVersion,
 		}
-		w, err := podClient.Watch(options)
-		Expect(err).NotTo(HaveOccurred(), "failed to set up watch")
+
+		listCompleted := make(chan bool)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				podList, err := podClient.List(options)
+				if err == nil {
+					select {
+					case listCompleted <- true:
+						framework.Logf("observed the pod list")
+					default:
+						listCompleted <- true
```

Comment:
I'm not familiar enough with the api machinery to effectively review, but this seems wrong, if the point of adding a `default:` was to avoid a blocked channel?

Reply:
correct. something like https://github.com/kubernetes/kubernetes/pull/73064#discussion_r258123083 should work

Reply:
@liggitt @spiffxp can you check now